lib/type/string.ex

```diff
@@ -36,6 +37,37 @@ defmodule Litmus.Type.String do
     end
   end
 
+  @spec convert(t, binary, map) :: {:ok, map}
+  defp convert(%__MODULE__{}, field, params) do
+    if Map.has_key?(params, field) and
+         !(is_binary(params[field]) or is_number(params[field]) or is_boolean(params[field])) do
+      {:error, "#{field} must be string"}
+    else
+      if is_binary(params[field]) do
+        {:ok, params}
+      else
+        modified_value = Kernel.inspect(params[field])
+        modified_params = Map.put(params, field, modified_value)
+        {:ok, modified_params}
+      end
+    end
+  end
+
+  @spec trim(t, binary, map) :: {:ok, map}
+  defp trim(%__MODULE__{trim: true}, field, params) do
+    if Map.has_key?(params, field) do
+      trimmed_value = String.trim(params[field])
+      trimmed_params = Map.put(params, field, trimmed_value)
+      {:ok, trimmed_params}
+    else
+      {:ok, params}
+    end
+  end
+
+  defp trim(%__MODULE__{trim: false}, _field, params) do
+    {:ok, params}
+  end
```

Comment:
Why did we move this up? I think we should keep our functions in alphabetical order when possible 👍